pkg/controller/jobframework/reconciler.go

```diff
@@ -243,6 +244,12 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 
 	// 1. make sure there is only a single existing instance of the workload.
 	// If there's no workload exists and job is unsuspended, we'll stop it immediately.
+	// We don't want to construct a workload for a job that has a parent job managed by kueue
+	// For example, a ray cluster created by RayJob
+	if owner := metav1.GetControllerOf(object); owner != nil && IsOwnerManagedByKueue(owner) {
+		return ctrl.Result{}, nil
+	}
```

Comment:
Let's try to make the implementation the same as for Job/JobSet.

In job_webhook.go, we add a parent workload annotation to the Job object.

We can do the same: add a parent workload annotation to the RayCluster if it's owned by a RayJob.